lib/cycle.js

```diff
@@ -111,7 +111,14 @@ exports.decycle = function decycle(object, options, replacer) {
           if (noCircularOption && withRefs) {
             return {$jsan: foundPath};
           }
-          if (path.indexOf(foundPath) === 0) {
+          
+          // This is only a true circular reference if the parent path is inside of foundPath
+          // drop the last component of the current path and check if it starts with foundPath
+          var pathElements = path.split('.');
+          var parentPathElements = pathElements.slice(0, pathElements.length - 1)
```

Comment:
nit: `pathElements.slice(0, -1)`

Reply:
Thanks, should be able to turn this into a one-liner if I'm not doing that.

Reply:
Cleaned this up a bit.